src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -160,14 +177,54 @@ public class RD2_OpportunityEvaluationService {
         return new List<Date>{ rd.npe03__Next_Payment_Date__c != null ? rd.npe03__Next_Payment_Date__c : Date.Today() };
     }
 
+    /**
+     * @description Extracts open installment Opportunities for the specified Recurring Donation
+     * @param rd Recurring donation containing list of all related Opportunities
+     * @return List<Opportunity> Open Opportunities
+     */
+    private List<Opportunity> extractOpenInstallmentOpps(npe03__Recurring_Donation__c rd) {
+        List<Opportunity> opps = new List<Opportunity>();
+
+        for (Opportunity opp : rd.npe03__Donations__r) {
+            if (opp.IsClosed == false) {
+                opps.add(opp);
+            }
+        }
+
+        return opps;
+    }
+
+    /**
+     * @description Updates, deletes or keep unchanged closed RD open Installment Opportunities based on the settings
+     * @param opps Opportunities
+     * @return void
+     */
+    private void handleClosedRecurringDonationOpps(List<Opportunity> opps) {
+        if (opps.isEmpty()) {
+            return;
+        }
+
+        if (rdSettings.npe03__Open_Opportunity_Behavior__c == CloseActions.Delete_Open_Opportunities.name()) {
+            delete opps;
+
+        } else if (rdSettings.npe03__Open_Opportunity_Behavior__c == CloseActions.Mark_Opportunities_Closed_Lost.name()) {
+            for (Opportunity opp : opps) {
```

Comment:
@force2b , should we create new SObject `new Opportunity(Id = opp.Id, StageName = System.Label.npe03.RecurringDonationClosedLostOpportunityStage);` for the update and not use other fields in the update DML?